packages/core/src/rest/SponsorBlock.ts

```diff
@@ -0,0 +1,60 @@
+import { VideoID, Category, Segment } from './SponsorBlock.types';
+import _ from 'lodash';
+
+const BASE_URL = 'https://sponsor.ajay.app';
+export const ALL_CATEGORIES = ['sponsor', 'intro', 'outro', 'interaction', 'selfpromo', 'music_offtopic'];
+
+export async function getSegments (videoID: VideoID, categories?: Category[]): Promise<Segment[]> {
+  let query = `?videoID=${videoID}`;
+
+  if (!categories) {
+    query += `&categories=${JSON.stringify(ALL_CATEGORIES)}`;
+  } else if (categories.length > 1) {
+    query += `&categories=${JSON.stringify(categories)}`;
+  } else if (categories.length === 1) {
+    query += `&category=${categories[0]}`;
+  } 
+
+  return new Promise((resolve) => {
+    try {
+      fetch(`${BASE_URL}/api/skipSegments${query}`)
+        .then(r => {
+          if (r.status !== 200) {
+            return resolve([]);
+          }
+          return r.json();
+        })
+        .then(j => {
+          if (!j) {
+            return resolve([]);
+          }
+
+          let segments = j.map(({category, segment}) => {
+            return {
+              category,
+              startTime: Math.round(segment[0]), 
+              endTime: Math.round(segment[1])
+            };
+          });
+
+          segments = _.sortBy(segments, ['startTime']);
+          const temp = _.concat(segments);
+          segments = _.filter(_.uniq(segments), (v: Segment) => {
```

Comment:
What segments are you trying to filter out here? Can you write a couple of tests to make sure it's doing what you intended?

Reply:
The api doesn't sort those segments in order of `startTime` so it will be not easy when comparing with `timePlayed`.
The purpose of filter is removing segment that is sub in another, like below:
<img width="429" alt="Screen Shot 2021-02-26 at 09 42 14" src="https://user-images.githubusercontent.com/12780733/109249692-cab0cf80-781a-11eb-9aa8-34a4693b2656.png">
I gonna add some tests